superset-frontend/src/components/ListView/ListView.tsx

```diff
@@ -292,9 +295,14 @@ function ListView<T extends object = any>({
 
   const cardViewEnabled = Boolean(renderCard);
   const [viewingMode, setViewingMode] = useState<ViewModeType>(
-    cardViewEnabled ? defaultViewMode : 'table',
+    cardViewEnabled ? viewMode || defaultViewMode : 'table',
```

Comment:
It looks like this is just duplicating the viewMode state that's now in the useListViewState hook. Can we remove this and just rely on `viewMode` and `setViewMode` instead?